web/modules/custom/asu_item_extras/src/Plugin/Block/ASUSearchItemIsPartOf.php

```diff
@@ -92,10 +92,10 @@ public function build() {
       $parent_node_id = $block_config['parent_node_id'];
       $first_title_view = $para->view(['type' => 'complex_title_formatter']);
       $parent_title = $this->renderer->render($first_title_view);
-      $parent_url = Url::fromRoute('entity.node.canonical', ['node' => $parent_node_id], ['absolute' => TRUE]);
-      $link = Link::fromTextAndUrl($parent_title, $parent_url)->toRenderable();
-      $rendered_link = render($link);
-      $output = 'Part of ' . $rendered_link;
+      $parent_entity = $this->entityTypeManager->getStorage('node')->load($parent_node_id);
```

Comment:
i'm concerned this will impact performance on search results if it has to load the parent node for each instance of an item on the page

Reply:
after discussion on this, we are deciding to rollback most of the changes with this branch and only use it for the media download links.